recipes/circularbuffer/all/conanfile.py

```diff
@@ -11,24 +17,32 @@ class CircularBufferConan(ConanFile):
     homepage = "https://github.com/rlogiacco/CircularBuffer"
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    def package_id(self):
-        self.info.header_only()
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def layout(self):
+        basic_layout(self)
 
     def package(self):
-        self.copy("LICENSE*", "licenses", self._source_subfolder)
-        self.copy("CircularBuffer.h", "include", self._source_subfolder)
-        self.copy("CircularBuffer.tpp", "include", self._source_subfolder)
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder,
+             "licenses"), src=self.source_folder)
+        copy(self, "CircularBuffer.h",
+             dst=os.path.join(self.package_folder, "include"), src=self.source_folder)
+        copy(self, "CircularBuffer.tpp",
+             dst=os.path.join(self.package_folder, "include"), src=self.source_folder)
+
+    def package_id(self):
+        self.info.clear()
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
```

Comment:
This recipe doesn't have `build()` step (header-only), so most probably doesn't require `generate` either.